scripts/enumerate_load_table_lookup.py

```diff
@@ -0,0 +1,377 @@
+import multiprocessing
+import sys
+import os
+import shutil, errno
+from pyspark.sql import SparkSession
+import pyspark.sql.functions as F
+import pyspark.sql.window as W
+from pyspark.sql.utils import AnalysisException
+import math
+import logging
+import functools
+import time
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+####################################################################
+### [[ SET UP SPARK SESSION ]] ###
+
+
+def init_spark1(name, mem="5gb", num_cpus=None):
+    """ Initialize a SparkSession on local machine """
+    if num_cpus is None:
+        num_cpus = multiprocessing.cpu_count()
+
+    cluster = os.environ.get("SPARK_CLUSTER")
+    if cluster is not None:
+        logger.info("Create SparkSession %s on existing cluster %s", name, cluster)
+        conf = SparkConf().setAppName(name).setMaster(cluster)
+        sc = SparkContext(conf=conf)
+        spark = (
+            SparkSession.builder.config(conf=conf)
+            .config("spark.executor.memory", mem)
+            .config("spark.cores.max", str(num_cpus))
```

Comment:
I've learned that this is max by default, so we should only set this when we want to cap CPU usage.